src/Storages/StorageInMemoryMetadata.cpp

```diff
@@ -345,6 +346,8 @@ Block StorageInMemoryMetadata::getSampleBlockForColumns(
         }
         else if (auto * it = virtuals_map.find(name); it != virtuals_map.end())
         {
+            if (skip_virtual_columns)
+                continue;
```

Comment:
This is not really how it should be fixed, becuase the same test case works for example with Values input format, and does not work with Parquet. So modifying StorageFile is not what is needed..